test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,257 @@
+		}
+		var err error
+		r.kclient, err = kubernetes.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create kubernetes client: %v", err))
+		}
+		r.mclient, err = pov1client.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create monitoring client: %v", err))
+		}
+		r.pclient = oc.NewPrometheusClient(tctx)
+		operatorConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		r.originalOperatorConfiguration = operatorConfiguration
+	})
+
+	g.AfterAll(func() {
+		currentConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		currentConfiguration.Data = r.originalOperatorConfiguration.Data
+		_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Update(tctx, currentConfiguration, metav1.UpdateOptions{})
+		o.Expect(err).To(o.BeNil())
+	})
+
+	g.It("should apply default collection profile initially", func() {
+		o.Eventually(func() error {
+			monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileDefault})
+			if err != nil {
+				return err
+			}
+			if len(monitors.Items) == 0 {
+				return fmt.Errorf("no monitors found with collection profile %q", collectionProfileDefault)
+			}
+
+			return nil
+		}).Should(o.BeNil())
+	})
+	g.It("should expose information about the applied collection profile using meta-metrics", func() {
+		vectorExpression := "profile:cluster_monitoring_operator_collection_profile:max{profile=\"%s\"} == 1"
+
+		for _, profile := range collectionProfilesSupportedList {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				queryResponse, err := helper.RunQuery(tctx, r.pclient, fmt.Sprintf(vectorExpression, profile))
+				if err != nil {
+					return err
+				}
+				if len(queryResponse.Data.Result) == 0 {
+					return fmt.Errorf("no result found for profile %q", profile)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		}
+	})
+	g.It("should have at-least one implementation for each collection profile", func() {
+		for _, profile := range collectionProfilesSupportedList {
+			err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+			o.Expect(err).To(o.BeNil())
+
+			o.Eventually(func() error {
+				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+				if err != nil {
+					return err
+				}
+				if len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", profile)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		}
+	})
+	g.It("should hide a default-only metric when minimal collection profile is enabled", func() {
+		defaultOnlyMetric := "kube_deployment_status_replicas"
```

Comment:
can we pick a Prometheus metric which we know for sure will (likely) never be present in the minimal profile (for instance `prometheus_engine_query_log_enabled`).